### DISCUSSION TITLE
Redownloading "missing" albums after folder rescan

### DISCUSSION THREAD

**Discussion by ndeast** [RandomNinjaAtk/docker-lidarr-extended#243] [Q&A] [answered]
Apologies if this is more a Lidarr issue than an extended issue, but I can't seem to figure it out. 

Anytime Lidarr goes through its folder rescan process it seems to take a long time for all of my albums to finish populating. So a bunch of albums that were previously at 100% look like this: 

![2023-05-22 10_14_43-American Football - Lidarr](https://github.com/RandomNinjaAtk/docker-lidarr-extended/assets/28495428/c7e4bc56-da6b-4018-ab40-506e091babb2)

Where a few tracks have yet to be picked up by Lidarr. These albums are then reported as "missing" and the lidarr extended script starts redownloading hundreds of albums that I already have. 

After a folder rescan, lidarr seems to go through the tags of every single file like so:

![2023-05-22 11_20_10-History - Lidarr](https://github.com/RandomNinjaAtk/docker-lidarr-extended/assets/28495428/ed8f14ad-97e6-4529-a315-446048ff8b21)

And this is what holds up the albums being seen as complete. 

My unmapped files is clear, I have lidarr-extended set to autoconfigure lidarr (although I have also tried disabling all of the auto rescans and watching the root folder to no avail). Write Metadata is set to All files; initial import only. Any help would be greatly appreciated.

**Comment by RandomNinjaAtk** [comment, marked as answer]
Seems like a general lidarr issue.  Probably better to take the issue upstream...

If you take your issue upstream, do not mention this project as they can be hostile... and your root issue appears to be a general lidarr problem.

It might be best for you to replicate the issue in one of the supported lidarr deployments...

Edit:
If you can't replicate the issue in a supported lidarr setup, that may indicate a problem here that may need to be investigated...

Keep in mind, this is the develop branch of lidarr and linuxservers docker container.

**Reply by ndeast** [reply]
Yeah that's what I was thinking. It happens just the same in my standard installation so I'll take it up there. 

Appreciate your response and work on the extended version. It's just fantastic.

**Reply by RandomNinjaAtk** [reply]
Honestly the other issue you'll have with support, is that I think they only run nightly. So you may have to upgrade to nightly to see if it's fixed.

**Comment by Yosh82** [comment]
Check you blocklist, if you cancelled a previous import, it could have added them to your blocklist, which will block the imports (I've had this issue before and found that was the cause).

Activity - Blocklist

**Comment by ndeast** [comment]
> Check you blocklist, if you cancelled a previous import, it could have added them to your blocklist, which will block the imports (I've had this issue before and found that was the cause).
> 
> 
> 
> Activity - Blocklist

Checked my block list, but it's empty. If I let it sit long enough after a scan, it will eventually fully import my whole library. It seems to want to rescan the meta data on every single track for some reason.

**Reply by Yosh82** [reply]
OK, thought I'd mention it incase it was that easy, rarely that lucky!

It's probably best to just let it do it's thing now it's running, but I've built a few installs over the years, and if just want a quick import and not to worry about retagging it all and renaming, fingerprinting etc, I usually disable all Lidarr file naming settings and fingerprinting, which is the one which takes a long time to run.
![image](https://github.com/RandomNinjaAtk/docker-lidarr-extended/assets/106485860/00d91408-c7f7-43da-84b7-6665f39fb31b)

And disable metadata settings, and untick scrub:
![image](https://github.com/RandomNinjaAtk/docker-lidarr-extended/assets/106485860/5741c8a5-ec32-44fd-ad51-0e6f83aa580e)

Everything imports near on instantly, then you can always amend the settings later and update the library 

But again, I'd just let it run now, or disable fingerprinting which causes the main delay, but may be less accurate at importing depending on the current metadata.

**Reply by RandomNinjaAtk** [reply]
Disabling rescan artist folder after artist refresh might help the situation. The only reason I'd see a need for you to rescan on refresh, would be if your manually importing and adding more files...

**Reply by ndeast** [reply]
Yeah I have all of my rescanning disabled after refresh, watch library folder disabled, and fingerprinting on new imports only. That works well. The problem is that folder rescan seems to be a 24 hour scheduled job that you can't (or at least I can't figure out how to) change. I took down my lidarr-extended container, and let my stock lidarr container (running release 1.1.4.3027) finish the rescan and a few hours later everything is once again imported, and my missing list is correct. 

I am gonna update my container to pull the nightly build and try a rescan again. I'll go bug the Lidarr people once I get that data. Again, appreciate both of your help.
